int32 emergency_stop_button
uint64 timestamp 
uint64 source_time
